From 22fc4ef525fd22a6ec0fbf535bfc87032db8d86b Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Wed, 23 Nov 2016 18:16:43 +0200
Subject: [PATCH 0620/1345] dts: a80x0: a70x0: a37x0: add customer board
 example device-tree

commit  5ff61c0e2e050bf415c4db03389f6b0bd155a5e4 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch adds the following device-trees:
	armada-8040-customer.dts
	armada-7040-customer.dts
	armada-3700-customer.dts

Those device-trees should be used only as an example for
a customer porting.
All interfaces are disabled, so using those device-trees is
not possible.

A costumer needs to modify the nodes according to the board
which is used, for example:
	- model: "Costumer board model"
	- compatible: "costumer-board-compatible"
	- MPPs: pinctrl-0 = <handle>
	- PHY addresses for network nodes
	- COMPHY handles for network nodes
	- SPI flash (if needed)
	- I2c devices (if needed)
	- Enable nodes

For more information refer to: ./Documentation/mvebu/porting-guide.txt

Change-Id: I27edc83d71c9a4333a4444312602498e714a55f5
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34011
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/Makefile               |    3 +
 .../boot/dts/marvell/armada-37x0-customer.dts      |  322 +++++++++++++++++++
 .../boot/dts/marvell/armada-70x0-customer.dts      |  334 ++++++++++++++++++++
 .../boot/dts/marvell/armada-80x0-customer.dts      |  306 ++++++++++++++++++
 4 files changed, 965 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-37x0-customer.dts
 create mode 100644 arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
 create mode 100644 arch/arm64/boot/dts/marvell/armada-80x0-customer.dts

diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index ece76b8..5a6eec9b 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -7,6 +7,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-sgmii1.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-sfp.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-community.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-37x0-customer.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-espressobin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-router.dtb
@@ -14,9 +15,11 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-nand.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-all-pcie.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-70x0-customer.dtb
 dtb-$(CONFIG_ARCH_MVEBU_PD) += armada-7040-pd.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-router.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-80x0-customer.dtb
 
 always		:= $(dtb-y)
 subdir-y	:= $(dts-dirs)
diff --git a/arch/arm64/boot/dts/marvell/armada-37x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-37x0-customer.dts
new file mode 100644
index 0000000..f0f6600
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-37x0-customer.dts
@@ -0,0 +1,322 @@
+/*
+ * Device Tree file for Marvell Armada 3720 development board
+ * (DB-88F3720-DDR3)
+ * Copyright (C) 2016 Marvell
+ *
+ * Gregory CLEMENT <gregory.clement@free-electrons.com>
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This file is distributed in the hope that it will be useful
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Example device tree for Armada 3700
+ * Customer may use this file as a reference.
+ * Each field that needs to be update (phys, mpps, addresses, frequencies)
+ * is marked with "UPDATE" comment.
+ *
+ * NOTE: all interfaces are disabled!
+ *
+ * Refer to ./Documentation/mvebu/porting_guide.txt for for information
+ */
+/dts-v1/;
+
+/* UPDATE
+ * Include one of the following dtsi files:
+ *	armada-372x.dtsi - used for dual core
+ *	armada-371x.dtsi - used for single core
+ */
+#include "armada-372x.dtsi"
+
+/ {
+	/* UPDATE */
+	model = "customer board model";
+	/* UPDATE
+	 * marvell,armada3720 - used for dual core
+	 * marvell,armada3710 - used for single core
+	 * NOTE: The compatile MUST match the included *.dtsi file
+	 */
+	compatible = "customer-board-compatible", "marvell,armada3720", "marvell,armada3710";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	/* UPDATE */
+	memory {
+		device_type = "memory";
+		reg = <0x00000000 0x00000000 0x00000000 0x20000000>;
+	};
+
+	soc {
+		internal-regs {
+			spi0: spi@10600 {
+				status = "disabled";
+				/* UPDATE */
+				spi-flash@0 {
+					#address-cells = <1>;
+					#size-cells = <1>;
+					compatible = "jedec,spi-nor";
+					reg = <0>; /* Chip select 0 */
+					spi-max-frequency = <104000000>;
+					m25p,fast-read;
+
+					pinctrl-names = "default";
+					pinctrl-0 = <&spi0_pins &spi_quad_pins>;
+
+					partition@0 {
+						label = "Bootloader";
+						reg = <0 0x200000>;
+					};
+					partition@200000 {
+						label = "U-boot Env";
+						reg = <0x200000 0x10000>;
+					};
+					partition@210000 {
+						label = "Linux";
+						reg = <0x210000 0xDF0000>;
+					};
+				};
+			};
+
+			i2c0: i2c@11000 {
+				status = "disabled";
+				/* UPDATE */
+				expander0: pca9555@22 {
+					compatible = "nxp,pca9555";
+					reg = <0x22>;
+					gpio-controller;
+					#gpio-cells = <2>;
+				};
+			};
+
+			i2c1: i2c@11080 {
+				status = "disabled";
+			};
+
+			/*
+			* Exported on the micro USB connector CON32
+			* through an FTDI
+			*/
+			uart0: serial@12000 {
+				status = "disabled";
+			};
+
+			eth0: ethernet@30000 {
+				/* UPDATE */
+				phy-mode = "rgmii-id";
+				status = "okay";
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
+			};
+
+			mdio: mdio@32004 {
+				/* UPDATE */
+				switch0: switch@0 {
+					compatible = "marvell,mv88e6085";
+					reg = <1>;
+
+					mdio@0 {
+						#address-cells = <1>;
+						#size-cells = <0>;
+
+						phy1: phy1@1 {
+							reg = <17>;
+						};
+						phy2: phy2@2 {
+							reg = <18>;
+						};
+						phy3: phy3@3 {
+							reg = <19>;
+						};
+					};
+				};
+			};
+
+			u3d@50000 {
+				status = "disabled";
+			};
+
+			udc@54100 {
+				status = "disabled";
+			};
+
+			/* CON31 */
+			usb3@58000 {
+				status = "disabled";
+				usb-phy = <&usb3_phy>;
+			};
+
+			usb@5e000 {
+				status = "disabled";
+			};
+
+			interrupt-controller@74840 {
+				status = "disabled";
+			};
+
+			sdhci1: sdhci@d0000 {
+				wp-inverted;
+				bus-width = <4>;
+				cd-gpios = <&gpio_sb 2 GPIO_ACTIVE_LOW>;
+				vqmmc-supply = <&vcc_sd_reg1>;
+				status = "disabled";
+				/* UPDATE */
+				pinctrl-names = "default";
+				pinctrl-0 = <&sdio_pins &cd_pins>;
+			};
+
+			sdhci0: sdhci@d8000 {
+				xenon,fixed-1-8v-pad-ctrl;
+				non-removable;
+				bus-width = <8>;
+				tuning-count = <9>;
+				status = "disabled";
+				/* UPDATE */
+				pinctrl-names = "default";
+				pinctrl-0 = <&mmc_pins>;
+			};
+
+			/* CON3 */
+			sata@e0000 {
+				target-supply = <&reg_5v_sata>;
+				status = "disabled";
+			};
+			vcc_sd_reg1: regulator@0 {
+				compatible = "regulator-gpio";
+				regulator-name = "vcc_sd1";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+
+				gpios = <&gpio_sb 23 GPIO_ACTIVE_HIGH>;
+				gpios-states = <0>;
+				states = <1800000 0x1 3300000 0x0>;
+				enable-active-high;
+			};
+		};
+
+		reg_sata: pwr-sata {
+			compatible = "regulator-fixed";
+			regulator-name = "pwr_en_sata";
+			regulator-min-microvolt = <12000000>;
+			regulator-max-microvolt = <12000000>;
+			enable-active-high;
+			regulator-boot-on;
+			gpio = <&expander0 2 GPIO_ACTIVE_HIGH>;
+		};
+
+		reg_5v_sata: v5-sata {
+			compatible = "regulator-fixed";
+			regulator-name = "v5.0-sata";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			vin-supply = <&reg_sata>;
+		};
+
+		reg_12v_sata: v12-sata {
+			compatible = "regulator-fixed";
+			regulator-name = "v12.0-sata";
+			regulator-min-microvolt = <12000000>;
+			regulator-max-microvolt = <12000000>;
+			vin-supply = <&reg_sata>;
+		};
+
+	};
+	dsa@0 {
+		compatible = "marvell,dsa";
+		dsa,ethernet = <&eth0>;
+		dsa,mii-bus = <&mdio>;
+		#address-cells = <2>;
+		#size-cells = <0>;
+
+		switch@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <1 0>;
+			/* UPDATE */
+			port@0 {
+				reg = <0>;
+				label = "cpu";
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+				label = "wan";
+				phy-handle = <&phy1>;
+			};
+
+			port@2 {
+				reg = <2>;
+				label = "lan0";
+				phy-handle = <&phy2>;
+			};
+
+			port@3 {
+			reg = <3>;
+				label = "lan1";
+				phy-handle = <&phy3>;
+			};
+		};
+	};
+	usb3_phy: usb3-phy {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&exp_usb3_vbus>;
+	};
+
+	exp_usb3_vbus: usb3-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb3-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		regulator-always-on;
+		gpio = <&expander0 1 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+/* CON17 (PCIe) / CON12 (mini-PCIe) */
+&pcie0 {
+	status = "disabled";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
new file mode 100644
index 0000000..ff6fa53
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
@@ -0,0 +1,334 @@
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Example device tree for Armada 7040
+ * Customer may use this file as a reference.
+ * Each field that needs to be update (phys, mpps, addresses, frequencies)
+ * is marked with "UPDATE" comment.
+ *
+ * NOTE: all interfaces are disabled!
+ *
+ * Refer to ./Documentation/mvebu/porting_guide.txt for for information
+ */
+
+/* UPDATE
+ * Include one of the following dtsi files:
+ *	armada-7040.dtsi - used for quad core
+ *	armada-7020.dtsi - used for dual core
+ */
+#include "armada-7040.dtsi"
+
+/ {
+	/* UPDATE */
+	model = "customer board model";
+	/* UPDATE
+	 * marvell,armada-ap806-quad - used for quad core
+	 * marvell,armada-ap806-dual - used for dual core
+	 * NOTE: The compatile MUST match the included *.dtsi file
+	 */
+	compatible = "customer-board-compatible", "marvell,armada7040",
+		     "marvell,armada-ap806-quad", "marvell,armada-ap806";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	aliases {
+		gpio1 = &gpio1;
+		gpio2 = &gpio2;
+	};
+
+	/* UPDATE */
+	memory@00000000 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+
+	ap806 {
+		config-space {
+			sdhci@6e0000 {
+				status = "disabled";
+			};
+			serial@512000 {
+				status = "disabled";
+			};
+			i2c@511000 {
+				status = "disabled";
+				/* UPDATE */
+				clock-frequency = <100000>;
+			};
+			sd_mmc_reg0: regulator@0 {
+			/* UPDATE */
+				gpios = <&gpio0 11 GPIO_ACTIVE_HIGH>;
+				gpios-states = <0>;
+				states = <1800000 0x1 3300000 0x0>;
+				enable-active-high;
+				status = "disabled";
+			};
+			sdhci@6e0000 {
+				/* UPDATE */
+				bus-width = <4>;
+				no-1-8-v;
+				non-removable;
+			};
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			serial@702000 {
+				status = "disabled";
+			};
+			serial@702100 {
+				status = "disabled";
+			};
+			serial@702200 {
+				status = "disabled";
+			};
+			serial@702300 {
+				status = "disabled";
+			};
+			sata: sata@540000 {
+				status = "disabled";
+			};
+			usb3h0: usb3@500000 {
+				status = "disabled";
+			};
+			usb3h1: usb3@510000 {
+				status = "disabled";
+			};
+			u3d@520000 {
+				/* UPDATE */
+				pinctrl-0 = <&u3d_vbus_pins>;
+				pinctrl-names = "default";
+				vbus-gpio = <&gpio2 30 GPIO_ACTIVE_HIGH>;
+				status = "disabled";
+			};
+			udc@524100 {
+				status = "disabled";
+			};
+			sdhci@780000 {
+				status = "disabled";
+			};
+			nand@720000 {
+				/* UPDATE */
+				nand-ecc-strength = <4>;
+				nand-ecc-step-size = <512>;
+				status = "disabled";
+				partition@0 {
+					label = "U-Boot";
+					reg = <0 0x200000>;
+				};
+				partition@200000 {
+					label = "Linux";
+					reg = <0x200000 0xd00000>;
+				};
+				partition@1000000 {
+					label = "Filesystem";
+					reg = <0x1000000 0x3f000000>;
+				};
+			};
+			spi@700680 {
+				status = "disabled";
+				/* UPDATE */
+				spi-flash@0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x1>;
+					compatible = "jedec,spi-nor";
+					reg = <0x0>;
+					spi-max-frequency = <20000000>;
+					partition@0 {
+						label = "boot";
+						reg = <0x0 0x200000>;
+					};
+					partition@200000 {
+						label = "Filesystem";
+						reg = <0x200000 0xd00000>;
+					};
+					partition@d00000 {
+						label = "boot_2nd";
+						reg = <0xf00000 0x100000>;
+					};
+				};
+			};
+			i2c1: i2c@701000 {
+				status = "disabled";
+				/* UPDATE */
+				clock-frequency = <100000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&i2c0_pins>;
+				eeprom@57 {
+					/* UPDATE */
+					compatible = "at24,24c64";
+					reg = <0x57>;
+					pagesize = <32>;
+				};
+				eeprom@50 {
+					/* UPDATE */
+					compatible = "at24,24c64";
+					reg = <0x50>;
+					pagesize = <32>;
+				};
+				expander0: pca9555@21 {
+					/* UPDATE */
+					compatible = "nxp,pca9555";
+					pinctrl-names = "default";
+					gpio-controller;
+					#gpio-cells = <2>;
+					reg = <0x21>;
+				};
+			};
+			i2c2: i2c@701100 {
+				status = "disabled";
+			};
+
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				/* UPDATE */
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					/* UPDATE */
+					phy-mode = "sgmii"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SGMII0>;
+					phy-names = "comphy";
+				};
+				emac2: mac2 {
+					/* UPDATE */
+					phy-mode = "sgmii"; /* lane-0 */
+					phy = <&phy2>;
+					phys = <&comphy0 0 COMPHY_SGMII1>;
+					phy-names = "comphy";
+				};
+				emac3: mac3 {
+					/* UPDATE */
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+				};
+			};
+			ppv22@000000 {
+				eth0: eth0@010000 {
+					status = "disabled";
+				};
+				eth1: eth1@020000 {
+					status = "disabled";
+				};
+				eth2: eth2@030000 {
+					status = "disabled";
+				};
+			};
+			eip197: eip197@800000 {
+				status = "disabled";
+			};
+			comphy {
+				status = "disabled";
+			};
+		};
+		pcie@0x600000 {
+			status = "disabled";
+			/* UPDATE */
+			num-lanes = <4>;
+		};
+		pcie@0x620000 {
+			status = "disabled";
+		};
+		pcie@0x640000 {
+			status = "disabled";
+		};
+	};
+
+	usb3h0_phy: usb3-h0-phy {
+		/* UPDATE */
+		compatible = "usb-nop-xceiv";
+		/* vcc-supply can be changed by io-expander or GPIO,
+		 * for A7040-DB we are using by default io-expander
+		 * for other board GPIO may be needed, should update
+		 * the phandle to gpio regulator
+		 */
+		vcc-supply = <&exp_usb3h0_vbus>;
+	};
+	usb3h1_phy: usb3-h1-phy {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&exp_usb3h1_vbus>;
+	};
+	exp_usb3h0_vbus: usb3-0-vbus {
+		compatible = "regulator-fixed";
+		/* UPDATE */
+		regulator-name = "usb3-0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		regulator-always-on;
+		gpio = <&expander0 0 GPIO_ACTIVE_HIGH>;
+	};
+	exp_usb3h1_vbus: usb3-1-vbus {
+		compatible = "regulator-fixed";
+		/* UPDATE */
+		regulator-name = "usb3-1-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		regulator-always-on;
+		gpio = <&expander0 1 GPIO_ACTIVE_HIGH>;
+	};
+	/* following GPIO entry is not used (for reference only)
+	 * GPIO is not connected in default A7040-DB board
+	 */
+	gpio_usb3_vbus: usb3-gpio-vbus {
+		compatible = "regulator-fixed";
+		/* UPDATE */
+		pinctrl-names = "default";
+		pinctrl-0 = <&xhci0_vbus_pins>;
+		regulator-name = "xhci0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		gpio = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+		status = "disabled";
+	};
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
new file mode 100644
index 0000000..5c8cf07
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
@@ -0,0 +1,306 @@
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Example device tree for Armada 8040
+ * Customer may use this file as a reference.
+ * Each field that needs to be update (phys, mpps, addresses, frequencies)
+ * is marked with "UPDATE" comment.
+ *
+ * NOTE: all interfaces are disabled!
+ *
+ * Refer to ./Documentation/mvebu/porting_guide.txt for for information
+ */
+
+/* UPDATE
+ * Include one of the following dtsi files:
+ *	armada-8040.dtsi - used for quad core
+ *	armada-8020.dtsi - used for dual core
+ */
+#include "armada-8040.dtsi"
+
+/ {
+	/* UPDATE */
+	model = "customer board model";
+	/* UPDATE
+	 * marvell,armada-ap806-quad - used for quad core
+	 * marvell,armada-ap806-dual - used for dual core
+	 * NOTE: The compatile MUST match the included *.dtsi file
+	 */
+	compatible = "customer-board-compatible", "marvell,armada8040",
+		     "marvell,armada-ap806-quad", "marvell,armada-ap806";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	aliases {
+		gpio1 = &gpio1;
+		gpio2 = &gpio2;
+	};
+
+	/* UPDATE */
+	memory@00000000 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+
+	ap806 {
+		config-space {
+			serial@512000 {
+				status = "disabled";
+			};
+			spi0: spi@510600 {
+				status = "disabled";
+				/* UPDATE */
+				spi-flash@0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x1>;
+					compatible = "jedec,spi-nor";
+					reg = <0x0>;
+					spi-max-frequency = <20000000>;
+					partition@0 {
+						label = "boot";
+						reg = <0x0 0x200000>;
+					};
+					partition@200000 {
+						label = "Filesystem";
+						reg = <0x200000 0xd00000>;
+					};
+					partition@d00000 {
+						label = "boot_2nd";
+						reg = <0xf00000 0x100000>;
+					};
+				};
+			};
+			i2c@511000 {
+				status = "disabled";
+				/* UPDATE */
+				clock-frequency = <100000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&ap_i2c0_pins>;
+				eeprom@57 {
+					/* UPDATE */
+					compatible = "at24,24c64";
+					reg = <0x57>;
+					pagesize = <32>;
+				};
+				eeprom@50 {
+					/* UPDATE */
+					compatible = "at24,24c64";
+					reg = <0x50>;
+					pagesize = <32>;
+				};
+			};
+
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			sata@540000 {
+				status = "disabled";
+			};
+			usb3h0: usb3@500000 {
+				status = "disabled";
+			};
+			usb3h1: usb3@510000 {
+				status = "disabled";
+			};
+			sdhci@780000 {
+				status = "disabled";
+			};
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				/* UPDATE */
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				/* UPDATE */
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					/* UPDATE */
+					phy-mode = "sfi"; /* lane-2 */
+					/* UPDATE */
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2: mac2 {
+					/* UPDATE */
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&phy2>;
+				};
+				emac3: mac3 {
+					/* UPDATE */
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+				};
+			};
+			ppv22@000000 {
+				/* UPDATE */
+				pinctrl-names = "default";
+				pinctrl-0 = <&cpm_ge2_rgmii_pins>; /* RGMII mpps */
+
+				eth0: eth0@010000 {
+					status = "disabled";
+				};
+				eth1: eth1@020000 {
+					status = "disabled";
+				};
+				eth2: eth2@030000 {
+					status = "disabled";
+				};
+			};
+			eip197@800000 {
+				status = "disabled";
+			};
+			comphy {
+				status = "disabled";
+			};
+			gpio1: gpio@440100 {
+			};
+			gpio2: gpio@440140 {
+			};
+			pinctrl@440000 {
+				/* UPDATE */
+				xhci0_vbus_pins: xhci0-vbus-pins {
+					marvell,pins = "mpp47";
+					marvell,function = "gpio";
+				};
+			};
+		};
+
+		pcie@0x640000 {
+			status = "disabled";
+		};
+		reg_usb3h0_vbus: usb3-vbus0 {
+			compatible = "regulator-fixed";
+			pinctrl-names = "default";
+			pinctrl-0 = <&xhci0_vbus_pins>;
+			regulator-name = "reg-usb3h0-vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			enable-active-high;
+			gpio = <&gpio2 15 GPIO_ACTIVE_HIGH>;
+			status = "okay";
+		};
+		vccq_sdio_reg: regulator@1 {
+			compatible = "regulator-fixed";
+			regulator-name = "vccq-sdio-reg";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+			status = "okay";
+		};
+
+	};
+	cpn-110-slave {
+		config-space {
+			usb3h0_1: usb3@500000 {
+				status = "disabled";
+			};
+			sata@540000 {
+				status = "disabled";
+			};
+			mdio@12a200 {
+				/* UPDATE */
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2_1: ethernet-phy@2 {
+					reg = <0>;
+				};
+			};
+			gop {
+				emac0_1: mac0 {
+					/* UPDATE */
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy1 2 COMPHY_SFI>;
+					phy-names = "disabled";
+				};
+				emac2_1: mac2 {
+					/* UPDATE */
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&phy2_1>;
+				};
+				emac3_1: mac3 {
+					/* UPDATE */
+					phy-mode = "sgmii"; /* lane-4 */
+				};
+			};
+			ppv22@000000 {
+				/* UPDATE */
+				pinctrl-names = "default";
+				pinctrl-0 = <&cps_ge1_rgmii_pins>;
+
+				eth0_1: eth0@010000 {
+					status = "disabled";
+				};
+				eth1_1: eth1@020000 {
+					status = "disabled";
+				};
+				eth2_1: eth2@030000 {
+					status = "disabled";
+				};
+			};
+			comphy {
+				status = "disabled";
+			};
+			nand@720000 {
+				status = "disabled";
+			};
+			tdm@7a0000 {
+				/* UPDATE */
+				pinctrl-0 = <&cps_tdm_pins>;
+				pinctrl-names = "default";
+				pclk-freq-mhz = <8>;
+				status = "disabled";
+			};
+		};
+		pcie@0x640000 {
+			status = "disabled";
+		};
+	};
+};
+
-- 
1.7.9.5

